### PR TITLE
#21563 - grant permission to admin

### PR DESCRIPTION
**Pull request by daniwe4**
Mantis 21563
grant permission to admin

### CODE REVIEW

File: Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php [hunk 1]
```diff
@@ -238,4 +238,20 @@ protected function cacheCheckAccessToObj($operation)
...
+	/**
+	 * Check wheter user is admin.
+	 *
+	 * @return bool
+	 */
+	public function isAdmin()
+	{
+		$global_roles = $this->review->assignedGlobalRoles($this->usr->getId());
+		foreach ($global_roles as $global_role) {
+			if(ilObject::_lookupTitle($global_role) == "Administrator") {
```
Comment (klees) [inline Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php]: Please do not use magic strings. In general. People might just rename that role and the stuff breaks.

Comment (klees) [reply Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php]: Ain't there something like "isSuperAdmin" on ilAccess, ilRBACReview or similar?

**Review by klees — CHANGES_REQUESTED** [review]
No magic strings, please.

File: Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php (hunk 1 above)
Comment (daniwe4) [reply Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php]: No, i didn´t find something in this direction. But i outsource the string into a constant.

Comment (lbruening) [reply Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php]: you could use 
```php
$rbacsystem->checkAccessOfUser($this->usr->getId(), "write", SYSTEM_FOLDER_ID)
```
i think, only admins has write access to this.... ("Administration" / "general settings")

Comment (mjansenDatabay) [reply Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php]: Why don't you check if the $global_role matches SYSTEM_ROLE_ID (global constant)?

Or even better: `$this->review->isAssigned($this->usr->getId(), SYSTEM_ROLE_ID);`

Comment (matthiaskunkel) [reply Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php]: Please consider that access permissions in Administration are also under RBAC control and that one can create new local and global roles with READ or WRITE permissions to any administration node. "Only admins have write access" doesn't go far enough.

File: Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php [hunk 2]
```diff
@@ -238,4 +239,20 @@ protected function cacheCheckAccessToObj($operation)
...
+	/**
+	 * Check wheter user is admin.
+	 *
+	 * @return bool
+	 */
+	public function isAdmin()
+	{
+		$global_roles = $this->review->assignedGlobalRoles($this->usr->getId());
+		foreach ($global_roles as $global_role) {
+			if(ilObject::_lookupTitle($global_role) == self::ADMIN) {
```
Comment (legionth) [inline Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php:252]: I would recommend to use an additional type check via `===`.

**Comment by klees** [comment]
Manually merged the changes with some additions today.

**Pull request closed without merging**